src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/Editor/DefaultTextBufferProjectService.cs

```diff
@@ -0,0 +1,136 @@
+
+        private readonly ITextDocumentFactoryService _documentFactory;
+        private readonly ErrorReporter _errorReporter;
+
+        public DefaultTextBufferProjectService(
+            ITextDocumentFactoryService documentFactory,
+            ErrorReporter errorReporter)
+        {
+            if (documentFactory == null)
+            {
+                throw new ArgumentNullException(nameof(documentFactory));
+            }
+
+            if (errorReporter == null)
+            {
+                throw new ArgumentNullException(nameof(errorReporter));
+            }
+
+            _documentFactory = documentFactory;
+            _errorReporter = errorReporter;
+        }
+
+        public override object GetHostProject(ITextBuffer textBuffer)
+        {
+            if (textBuffer == null)
+            {
+                throw new ArgumentNullException(nameof(textBuffer));
+            }
+
+            // If there's no document we can't find the FileName, or look for an associated project.
+            if (!_documentFactory.TryGetTextDocument(textBuffer, out var textDocument))
+            {
+                return null;
+            }
+
+            var projectsContainingFilePath = IdeApp.Workspace.GetProjectsContainingFile(textDocument.FilePath);
+            foreach (var project in projectsContainingFilePath)
+            {
+                if (!(project is DotNetProject))
+                {
+                    continue;
+                }
+
+                var projectFile = project.GetProjectFile(textDocument.FilePath);
+                if (!projectFile.IsHidden)
+                {
+                    return project;
+                }
+            }
+
+            return null;
+        }
+
+        public override string GetProjectPath(object project)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            var dotnetProject = project as DotNetProject;
+
+            Debug.Assert(dotnetProject != null);
+
+            return dotnetProject.FileName.FullPath;
+        }
+
+        public override bool IsSupportedProject(object project)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            var dotnetProject = (DotNetProject)project;
+
+            Debug.Assert(dotnetProject != null);
+
+            try
+            {
+                // ASP.NET Core Web projects
+                if (dotnetProject.IsCapabilityMatch(AspNetCoreWebCapability))
+                {
+                    return true;
+                }
+
+                // ASP.NET Core Class libraries
+                if (dotnetProject.References.Any(reference => reference.Reference.StartsWith(MvcAssemblyName, StringComparison.Ordinal)))
+                {
+                    return true;
+                }
```

Comment:
Do we need this at all? Why not just `return true`. There no non-Core Razor in VS4Mac

Reply:
We could. Probably being a little bit overprotective here.